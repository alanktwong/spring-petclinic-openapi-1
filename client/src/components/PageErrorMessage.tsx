import * as React from 'react';
import { IError } from '../types';

const PageErrorMessage: React.FunctionComponent<{ error?: IError }> = ({
  error,
}) => {
  if (!error) {
    return <></>;
  }
  return (
    <div role="alert" className="alert alert-warning">
      <strong>Error!</strong> {error.message}
    </div>
  );
};

export default PageErrorMessage;
