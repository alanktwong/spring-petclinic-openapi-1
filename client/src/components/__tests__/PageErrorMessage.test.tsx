import React from 'react';
import renderer from 'react-test-renderer';
import PageErrorMessage from '../PageErrorMessage';

it('renders correctly', () => {
  const tree = renderer
    .create(<PageErrorMessage error={{ message: 'msg1' }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
