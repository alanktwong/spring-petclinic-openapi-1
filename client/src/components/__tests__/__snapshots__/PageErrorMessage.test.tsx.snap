// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="alert alert-warning"
  role="alert"
>
  <strong>
    Error!
  </strong>
   
  msg1
</div>
`;
